mantidproject/mantid
# Add units for fit parameters to model fitting

**Description of work.**
This PR changes Model fitting to try and get the units for its plots from a dictionary of parameters and log names. This is current a limited list but can be expanded easily if a log or parameter is widly used with the same unit.

**Report to:** James Lord

**To test:**

1. Open Muon analysis
2. Load MUSR 62260
3. In the fitting tab add a GausOsc function
4. In Sequential fitting tab click `Sequentialy fit all`
5. In the Results tab check sample_temp and sample_magn_field in Log values
6. Output results table.
7. Go to the model fitting tab
8. Change the selected data in the X axis to different parameters and check that in the plot the units for, sample_temp, sample_magn_field, sigma, Frequency and Phi are correct.
9. Repeat with the y axis.

Fixes #32453 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
